### PR TITLE
Dryrun as preparelocal

### PR DESCRIPTION
**Pull request by novicecpp**

### CODE REVIEW

File: src/python/CRABClient/ClientUtilities.py [hunk 1]
```diff
@@ -788,7 +788,7 @@ def checkStatusLoop(logger, server, api, taskname, targetstatus, cmdname):
     logger.debug("Ended %s process." % ("resubmission" if cmdname == "resubmit" else "submission"))
 
 
-def execute_command(command=None, logger=None, timeout=None, redirect=True):
+def execute_command(command=None, logger=None, timeout=None, redirect=True, live=False):
```
Comment (belforte) [inline src/python/CRABClient/ClientUtilities.py:791]: what does live do "in English" ?
can you help me remember what "redirect" does ?
where did you plan to use "live" ?

File: src/python/CRABClient/Commands/preparelocal.py [hunk 2]
```diff
@@ -65,10 +68,19 @@ def getInputFiles(self):
         crabDBInfo, _, _ = server.get(api='task', data={'subresource': 'search', 'workflow': taskname})
         status = getColumn(crabDBInfo, 'tm_task_status')
         self.destination = getColumn(crabDBInfo, 'tm_asyncdest')
+        username = getColumn(crabDBInfo, 'tm_username')
+        sandboxName = getColumn(crabDBInfo, 'tm_user_sandbox')
 
         inputsFilename = os.path.join(os.getcwd(), 'InputFiles.tar.gz')
+        sandboxFilename = os.path.join(os.getcwd(), 'sandbox.tar.gz')
         if status == 'UPLOADED':
```
Comment (belforte) [inline src/python/CRABClient/Commands/preparelocal.py:76]: do we still need this "if" ? We could always pull files from S3, instead of scheduler.

File: src/python/CRABClient/Commands/recover.py
Comment (belforte) [inline src/python/CRABClient/Commands/recover.py:1]: why is this file changed ? DId you do something, or simply old version from the time you created your branch ?

File: src/python/CRABClient/ClientUtilities.py (hunk 1 above)
Comment (novicecpp) [reply src/python/CRABClient/ClientUtilities.py:791]: Ah...my bad.
It should be `livelog`: print log as soon as it arrive from subcommand (I plan to use logger instead of print).
`redirect` is you want to capture subprocess stdout/stderr into python variable (available in `proc.communicate()` line below).

File: src/python/CRABClient/Commands/preparelocal.py (hunk 2 above)
Comment (novicecpp) [reply src/python/CRABClient/Commands/preparelocal.py:76]: Good catch. 
It should be `if status in ['UPLOADED', 'SUBMITTED']:`.

File: src/python/CRABClient/Commands/recover.py
Comment (novicecpp) [reply src/python/CRABClient/Commands/recover.py:1]: Yeah, the old version. No change on this file.

File: src/python/CRABClient/ClientUtilities.py (hunk 1 above)
Comment (belforte) [reply src/python/CRABClient/ClientUtilities.py:791]: after some testing we think that we do not need `live` option and can simply set `redirect=False`

File: src/python/CRABClient/Commands/preparelocal.py (hunk 2 above)
Comment (belforte) [reply src/python/CRABClient/Commands/preparelocal.py:76]: OK

File: src/python/CRABClient/Commands/recover.py
Comment (belforte) [reply src/python/CRABClient/Commands/recover.py:1]: OK I will revert

**Comment by belforte** [comment]
implemented slightly differently via #5348
